ipywidgets/widgets/widget.py

```diff
@@ -483,7 +483,7 @@ def _compare(self, a, b):
         else:
             return a == b
 
-    def _compare_buffers(self, a, b, key):
+    def _buffer_list_equal(self, a, b):
```

Comment:
> I mean extension point as in "something an inheriting class might want to override".

Ah. Since it depends on the implementation of the top-level functions, I think it makes more sense to not override it, and instead put it at the top level too.